core/src/client/async_client/mod.rs

```diff
@@ -145,64 +145,36 @@ impl ThreadSafeRequestManager {
 		self.0.lock().expect("Not poisoned; qed")
 	}
 }
+
+pub(crate) type SharedDisconnectReason = Arc<AsyncRwLock<Option<Arc<Error>>>>;
```

Comment:
Do we need the inner arc in the `Arc<Error>`?

Reply:
The error is not clone so this a just hack around that, any better ideas?

Reply:
Ah ok, in that case it's all gravy :)